### ISSUE TITLE
Picker: add event after value is updated / modal closed

### ISSUE THREAD

**Issue by m-vo** [contao/contao#464]
**Description**  
I'd like to (visually) enhance the results of a picker selection - e.g. by resolving the title or other attributes for a list of ids and displaying them in the DOM. It's however really hard to get notified on when the value of the underlying text field is changing because it is set programmatically ( `field.value = ...`) and therefore no DOM events are happening. 

What about adding a custom event that gets triggered as soon as the value gets set (usually when the user hits a button / modal window closes again)?

**Current workaround**  
There is an ugly workaround that involves redefining the value property but it also destroys a lot of DOM properties. It roughly works like this:
```js
Object.defineProperty(myInputField, "value", {
    get: function () {
        return this._value;
    },
    set: function (v) {
        this._value = v;
        onValueUpdated(v);
    }
});

// ...

function restoreNode(node, input) {
    node.value = input.value;
    input.parentNode.replaceChild(node, input);
}

const nodeBackup = myInputField.cloneNode(false);
document.querySelector('.tl_edit_form').addEventListener(
    'submit', restoreNode.bind(null, nodeBackup, myInputField)
);
```

**Example**
It would be nice if I could do something like this:
```js
myInputField.addEventListener('valueUpdated', e => ...));
```

Does anyone see another solution? :smile:

**Comment by asaage** [comment]
The modal/picker should probably just fire (trigger) the ordinary ```'change'``` Event of its underlying field.

**Comment by leofeyer** [comment]
@contao/developers Is this a bugfix or a new feature?

**Comment by Toflar** [comment]
Feature.

**Comment by leofeyer** [comment]
Implemented in be4aeccc9d794cd6c0aa768f8c3d52be8c7efd06.
